### PR TITLE
Partial client package completion - 1.8.9

### PR DESCRIPTION
**Pull request by mciolkosz**
The `client` base package, the `gui` sub-package and all other sub-packages of that, and then anything alphabetically between the 2.

Side question: Are merges normally squashed, or not?

### CODE REVIEW

**Comment by BoogieMonster1O1** [comment]
merges are normally squashed. feel free to pollute your commit history lol

File: mappings/net/minecraft/block/material/Material.mapping [hunk 1]
```diff
@@ -23,11 +23,11 @@ CLASS net/minecraft/class_63 net/minecraft/block/material/Material
 	FIELD field_131 FOILAGE Lnet/minecraft/class_63;
 	FIELD field_132 PLANT Lnet/minecraft/class_63;
 	FIELD field_133 REPLACEABLE_PLANT Lnet/minecraft/class_63;
-	FIELD field_134 CHEESE Lnet/minecraft/class_63;
+	FIELD field_134 SPONGE Lnet/minecraft/class_63;
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/block/material/Material.mapping:26]: cheese

File: mappings/net/minecraft/client/gui/screen/Screen.mapping [hunk 2]
```diff
@@ -57,7 +57,7 @@ CLASS net/minecraft/class_388 net/minecraft/client/gui/screen/Screen
 		ARG 1 button
 	METHOD method_14502 (Lnet/minecraft/class_1071;)Ljava/util/List;
 		ARG 1 stack
-	METHOD method_4259 mouseDragged (IIIJ)V
+	METHOD method_4259 mouseX (IIIJ)V
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/client/gui/screen/Screen.mapping:60]: this doesnt sound right

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/material/Material.mapping (hunk 1 above)
Comment (mciolkosz) [reply mappings/net/minecraft/block/material/Material.mapping:26]: Gonna ask: why `cheese`?

File: mappings/net/minecraft/client/gui/screen/Screen.mapping (hunk 2 above)
Comment (mciolkosz) [reply mappings/net/minecraft/client/gui/screen/Screen.mapping:60]: It isn't, it's what happens when I mis-click on something while tired

File: mappings/net/minecraft/block/material/Material.mapping (hunk 1 above)
Comment (mciolkosz) [reply mappings/net/minecraft/block/material/Material.mapping:26]: Also, was already approved here: https://github.com/Legacy-Fabric/yarn/pull/255/files#

The only reason it's popping up again is cuz of the squash merging, and I had already started the `client` stuff before rebasing my fork.

File: mappings/net/minecraft/client/gui/screen/Screen.mapping (hunk 2 above)
Comment (mciolkosz) [reply mappings/net/minecraft/client/gui/screen/Screen.mapping]: Feexed

**Comment by mciolkosz** [comment]
Anything else that I'm missing?

**Comment by mciolkosz** [comment]
_nudge_

**Review by BleachDev — APPROVED** [review]

**Review by PrateekBing — APPROVED** [review]

**Review by heftymouse — APPROVED** [review]
:shipit:

**Review by not-good-keeper — APPROVED** [review]
lgtm

**Review by Mysticpasta1 — APPROVED** [review]

**Review by srujandeshpande — APPROVED** [review]
Lgtm

**Review by FokshaWasTaken — APPROVED** [review]

**Comment by FokshaWasTaken** [comment]
LGTM

**Review by LePichu — APPROVED** [review]
LGTM!

File: mappings/net/minecraft/client/sound/SoundCategory.mapping [hunk 3]
```diff
@@ -26,3 +26,7 @@ CLASS net/minecraft/class_2153 net/minecraft/client/sound/SoundCategory
 	METHOD method_8383 byName (Ljava/lang/String;)Lnet/minecraft/class_2153;
 		ARG 0 name
 	METHOD method_8384 getId ()I
+	METHOD values values ()[Lnet/minecraft/class_2153;
+	METHOD values values ()[Lnet/minecraft/class_2153;
+	METHOD values values ()[Lnet/minecraft/class_2153;
+	METHOD values values ()[Lnet/minecraft/class_2153;
```
Comment (modmuss50) [inline mappings/net/minecraft/client/sound/SoundCategory.mapping:32]: Broken, remove

File: mappings/net/minecraft/client/sound/SoundSystem.mapping [hunk 4]
```diff
@@ -51,23 +51,130 @@ CLASS net/minecraft/class_622 net/minecraft/client/sound/SoundSystem
...
 	METHOD method_7109 tick ()V
 	METHOD method_7110 pauseAll ()V
 	METHOD method_7111 resumeAll ()V
 	METHOD method_7114 start ()V
 	CLASS 2
-		METHOD openConnection (Ljava/net/URL;)Ljava/net/URLConnection;
+		METHOD openConnection openConnection (Ljava/net/URL;)Ljava/net/URLConnection;
 			ARG 1 url
+		CLASS 1
+			METHOD connect connect ()V
```
Comment (modmuss50) [inline mappings/net/minecraft/client/sound/SoundSystem.mapping:76]: Wat? Remove all of this

**Comment by arthurbambou** [comment]
This is definitely the most review I saw on a pr on this repo 😄

File: mappings/net/minecraft/realms/RealmsButton.mapping [hunk 5]
```diff
@@ -14,6 +23,15 @@ CLASS net/minecraft/realms/RealmsButton net/minecraft/realms/RealmsButton
...
 	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
+	METHOD active isActive ()Z
```
Comment (modmuss50) [inline mappings/net/minecraft/realms/RealmsButton.mapping:34]: Why so many duplicate mappings, remove

File: mappings/net/minecraft/realms/RealmsButton.mapping [hunk 6]
```diff
@@ -26,10 +44,46 @@ CLASS net/minecraft/realms/RealmsButton net/minecraft/realms/RealmsButton
...
+		ARG 2 mouseY
+	METHOD clicked mouseClicked (II)V
+		ARG 1 mouseX
+		ARG 2 mouseY
+	METHOD clicked mouseClicked (II)V
+		ARG 1 mouseX
+		ARG 2 mouseY
+	METHOD clicked mouseClicked (II)V
+		ARG 1 mouseX
+		ARG 2 mouseY
```
Comment (modmuss50) [inline mappings/net/minecraft/realms/RealmsButton.mapping:73]: Again remove all these duplicates, this is a mess

File: mappings/net/minecraft/realms/RealmsButton.mapping [hunk 7]
```diff
@@ -26,10 +44,46 @@ CLASS net/minecraft/realms/RealmsButton net/minecraft/realms/RealmsButton
...
-	METHOD getYImage (Z)I
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
+	METHOD getProxy getButton ()Lnet/minecraft/class_356;
```
Comment (modmuss50) [inline mappings/net/minecraft/realms/RealmsButton.mapping:84]: Fix

File: mappings/net/minecraft/realms/RealmsButton.mapping [hunk 8]
```diff
@@ -41,6 +95,33 @@ CLASS net/minecraft/realms/RealmsButton net/minecraft/realms/RealmsButton
 	METHOD render (II)V
 		ARG 1 mouseX
 		ARG 2 mouseY
-	METHOD renderBg (II)V
+	METHOD render render (II)V
```
Comment (modmuss50) [inline mappings/net/minecraft/realms/RealmsButton.mapping:98]: Fix

**Comment by FokshaWasTaken** [comment]
Actually, for those who may be concerned, I highly suggest reading this conversation on the official Fabric Discord

<https://discord.com/channels/507304429255393322/789953494534324264/1052984828494094438>

File: mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping [hunk 9]
```diff
@@ -1,27 +1,110 @@
 CLASS net/minecraft/realms/RealmsClickableScrolledSelectionList net/minecraft/realms/RealmsClickableScrolledSelectionList
 	FIELD proxy list Lnet/minecraft/class_2311;
+	FIELD proxy widget Lnet/minecraft/class_2311;
+	FIELD proxy widget Lnet/minecraft/class_2311;
+	FIELD proxy widget Lnet/minecraft/class_2311;
+	FIELD proxy widget Lnet/minecraft/class_2311;
+	FIELD proxy widget Lnet/minecraft/class_2311;
+	FIELD proxy widget Lnet/minecraft/class_2311;
```
Comment (modmuss50) [inline mappings/net/minecraft/realms/RealmsClickableScrolledSelectionList.mapping:8]: Fix

**Review by modmuss50 — CHANGES_REQUESTED** [review]
Something is seriously wrong with your mappings.

**Comment by mciolkosz** [comment]
I personally didn't notice the files being filled with duplicates, I would update the class, click away to something else, get buried in it, and then apparently go and find my way back into the class, and re-update what was once again not marked as deobfuscated.

I just tried it now with `net/minecraft/realms/RealmsButton`, updating `#getProxy` to `getButton`. Save and close the file, then close Enigma. Reopen Enigma, and it's like it never saved it (marked as deobf again), but the mapping is saved in the file.

**Pull request merged**